tests/PHPStan/Analyser/data/isset.php

```diff
@@ -10,16 +10,16 @@ class Foo
 	 */
 	public function doFoo(array $integers, string $string, $mixedIsset, $mixedArrayKeyExists)
 	{
-		if (rand(0, 1) === 0) {
+		if (rand(0, 10) === 0) {
```

Comment:
It seems that PhpStan already has something definition of pure functions and therefore using rand with same args is cached. 🤔 But not sure what is the current state.

Reply:
Actually not yet, it currently dumbly caches `rand(0, 1)` result for all functions.